isok-cli/src/new_agent.rs

```diff
@@ -0,0 +1,91 @@
+use std::{io, time::SystemTime};
+
+use biscuit_auth::{builder_ext::BuilderExt, Biscuit, KeyPair, PrivateKey};
+use chrono::DateTime;
+
+use super::private_key;
+
+fn parse_ttl(s: &str) -> Result<SystemTime, chrono::ParseError> {
+    // NOTE: unlike biscuit we don't currently support specifying a duration here
+    Ok(DateTime::parse_from_rfc3339(s)?.to_utc().into())
+}
+
+fn parse_service(s: &str) -> io::Result<Box<str>> {
+    let s = s.trim();
+
+    if s.is_empty() {
+        return Err(io::Error::new(
+            io::ErrorKind::InvalidInput,
+            "service name cannot be empty",
+        ));
+    }
+
+    Ok(s.to_owned().into_boxed_str())
+}
+
+/// Generate a biscuit token for a new agent.
+#[derive(Debug, Clone, clap::Parser)]
+pub struct NewAgentCommand {
+    /// Private of the broker, used to sign the token.
+    #[clap(long, value_name = "KEY",  value_parser = private_key::parse,)]
```

Comment:
Add `short` derive param, in order to do `command -p KEY`. In this context it facilitates the CLI
```suggestion
    #[clap(short, long, value_name = "KEY",  value_parser = private_key::parse,)]
```